File: PROGRAMMERS/카펫.js
function solution(brown, yellow) {
    const candidate = [];
    const add = brown + yellow;
    for(let i = 3; i<= Math.sqrt(add) >> 0; i++){
        if(add % i ===0) candidate.push([add / i, i]);
    }
    for(const [width, height] of candidate){
        if((height - 2) * (width -2) === yellow) return [width , height];
    }
    return candidate[0];
}

File: PROGRAMMERS/카펫.py
from math import floor, sqrt 
def solution(brown, yellow):
    total = brown + yellow
    for i in range(2, floor(sqrt(total)) + 1):
        if total % i != 0:
            continue
        width = total / i - 2
        height = i - 2
        if width * height == yellow:
            return [total / i, i] 
​